gc/gc-base/src/main/java/org/projectnessie/gc/base/GCParams.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.gc.base;
+
+import java.io.Serializable;
+import java.time.Instant;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.immutables.value.Value;
+
+/** Config params for GC. */
+@Value.Immutable
+public interface GCParams extends Serializable {
+
+  /** Nessie client configurations from {@link org.projectnessie.client.NessieConfigConstants}. */
+  Map<String, String> getNessieClientConfigs();
+
+  /** Optional cutoff time per live reference. */
+  @Nullable
+  Map<String, Instant> getCutOffTimestampPerRef();
+
+  /** Default cutoff time for all the references. */
+  Instant getDefaultCutOffTimestamp();
+
+  /** Optional cutoff time for all the dead references. */
+  @Nullable
+  Instant getDeadReferenceCutOffTimeStamp();
+
+  /**
+   * Optional spark partitions count to be used for distributing references. Default total reference
+   * count (live + dead) will be used.
+   */
+  @Nullable
+  Integer getSparkPartitionsCount();
+
+  /**
+   * Commit protection duration in hours to avoid expiring on going or recent commits. Default is 2
+   * hours.
+   */
+  @Value.Default
+  default int getCommitProtectionDuration() {
```

Comment:
are we not able to return `Duration` here? 
If not, I think it should be renamed to `getCommitProtectionHours`.
Now, the caller needs to look into implementation to see that the unit is hours.

Reply:
(Mentioned it above) it should be a `Duration`, which would fit well with `Instant`